### PR TITLE
Endpoints for QA Cleanup Suite

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/metadata/routes/comments.clj [hunk 1]
```diff
@@ -5,6 +5,37 @@
...
+                  :default {:schema      ErrorResponse
+                            :description "All other errors"}}
+      :summary "List All Comments Added by a User"
+      :description "This endpoint allows an administrator to list all comments that were entered by a single user."
+      (ok (comments/list-user-comments commenter-id)))
+
+    (DELETE "/:commenter-id" []
+      :path-params [commenter-id :- CommenterId]
+      :query [{:keys [user]} StandardUserQueryParams]
+      :responses {200      {:schema      (describe String "Empty response")
```
Comment (slr71) [inline src/metadata/routes/comments.clj]: This is kind of hokey, but it was the only way that I found to get compojure-api not to complain about the empty response when using this type of endpoint definition.

File: src/metadata/routes/tags.clj [hunk 2]
```diff
@@ -10,6 +10,33 @@
...
+                  :default {:schema      ErrorResponse
+                            :description "All other errors"}}
+      :summary "List All Attached Tags for a User"
+      :description
+      "This endpoint lists all tags that have been attached to a file or folder by a user."
+      (ok (tags/list-all-attached-tags user)))
+
+    (DELETE "/tags" []
+      :query [{:keys [user]} StandardUserQueryParams]
+      :responses {200      {:schema      (describe String "Empty response")
```
Comment (slr71) [inline src/metadata/routes/tags.clj]: Another case where I couldn't find a better way to get compojure-api to handle this.

File: src/metadata/routes/tags.clj [hunk 3]
```diff
@@ -66,6 +93,33 @@ Given a textual fragment of a tag's value, this endpoint will list up to a given
...
+                  :default {:schema      ErrorResponse
+                            :description "All other errors"}}
+      :summary "List Tags Defined by a User"
+      :description
+      "This endpoint lists all of the tags defined by a user."
+      (ok (tags/list-tags-defined-by user)))
+
+    (DELETE "/user" []
+      :query [{:keys [user]} StandardUserQueryParams]
+      :responses {200      {:schema      (describe String "Empty response")
```
Comment (slr71) [inline src/metadata/routes/tags.clj]: And another...

File: src/metadata/services/tags.clj [hunk 4]
```diff
@@ -127,7 +164,7 @@
      limit - The `limit` query parameter. It should be a positive integer."
   [user contains limit]
   (let [matches (db/get-tags-by-value user (str "%" contains "%") limit)]
-    {:tags (map #(dissoc % :owner_id) matches)}))
+    {:tags matches}))
```
Comment (slr71) [inline src/metadata/services/tags.clj]: The owner ID isn't actually returned by the query, so I just deleted the code that was being used to remove it.

File: src/metadata/routes/comments.clj (hunk 1 above)
Comment (psarando) [reply src/metadata/routes/comments.clj]: The `DELETE /user/:tag-id` endpoint is using something like this:
```clojure
200 {:schema      nil
     :description "The tag was successfully deleted"}
```
Would that work with the `DELETE` endpoints you added? Except the responses would probably have to be just `(ok)`. 👌

File: src/metadata/services/comments.clj [hunk 5]
```diff
@@ -16,6 +17,10 @@
       (dissoc :retracted_by)
       (assoc :post_time (.getTime (:post_time comment)))))
 
+(defn- format-comment-details
+  [comment]
+  (remove-nil-values (update comment :post_time #(.getTime %))))
```
Comment (psarando) [inline src/metadata/services/comments.clj]: 👍

File: src/metadata/services/favorites.clj [hunk 6]
```diff
@@ -43,13 +56,22 @@
      entity-type - This is the value of the `entity-type` query parameter. It should be a string
                    containing one of the following: any|file|folder."
   [user entity-type]
-  (let [entity-types (if (= "any" entity-type)
-                       ["file" "folder"]
-                       [entity-type])]
+  (let [entity-types (determine-filesystem-entity-types entity-type)]
```
Comment (psarando) [inline src/metadata/services/favorites.clj]: 👍

File: src/metadata/routes/comments.clj (hunk 1 above)
Comment (slr71) [reply src/metadata/routes/comments.clj]: Ah! I finally spotted the difference!

```clojure
       :coercion middleware/no-response-coercion
```

Adding this allowed me to return a truly empty response.

**Comment by slr71** [comment]
Thanks for the review!

File: src/metadata/routes/comments.clj (hunk 1 above)
Comment (psarando) [reply src/metadata/routes/comments.clj]: Ah, yeah, I forgot about that requirement too 😖

**Pull request merged**